### PR TITLE
feat: add session Jupyter free images - Renku 2.0

### PR DESCRIPTION
**Pull request by andre-code**
PR for Jupyter free images - Renku 2.0

- [x] Support custom environment variables in Renku 2.0 - [#3227](https://github.com/SwissDataScienceCenter/renku-ui/pull/3227)
- [x] Use the data-service sessions endpoint - [#3296](https://github.com/SwissDataScienceCenter/renku-ui/pull/3296)
- [x] Resolve issues with data connectors 
- [x] Display an error message for session launch failures
- [x] Show an error when the image doesn't exist during session launch 

/deploy renku=feat-jupyter-free-sessions renku-data-services=main amalthea-sessions=main renku-notebooks=master extra-values=amalthea-sessions.deployCrd=false

### CODE REVIEW

**Review by leafty — COMMENTED** [review]
It is possible that the double POST bug is present in this PR:

1. POST with empty list of data connectors
2. POST with data connectors

**Review by leafty — COMMENTED** [review]
Screenshot 2024-10-30 at 14 40 30

Rename "fragment" to "path" (or "pathname"), URL component have specific names we have to adhere to -> `URI = scheme ":" ["//" authority] path ["?" query] ["#" fragment]`. See also: https://developer.mozilla.org/en-US/docs/Web/API/URL.

Also, the UI is not respecting this setting. The `<iframe>` needs to contain the URL path setting, otherwise it cannot support JupyterLab. I can open the interface with "Open in new tab" and manually adding the path there.

**Comment by andre-code** [comment]
Thank you @leafty, for reviewing the PR and reporting the errors:

Visual bug: Fixed
Double POST issue: Fixed
Error in showing the session or opening the URL link: I'm holding this for now to see if Tasko can include the fix in the data service; otherwise, I can retrieve the default URL from the launcher and make it work.

**Comment by andre-code** [comment]
> > Error in showing the session or opening the URL link: I'm holding this for now to see if Tasko can include the fix in the data service; otherwise, I can retrieve the default URL from the launcher and make it work.
> 
> ~Why does this need a backend fix? It is purely a UI issue.~ Nevermind.

@leafty Tasko will add the defaultURL to the backend GET /sessions response

File: client/src/components/Logs.tsx [hunk 1]
```diff
@@ -353,38 +367,24 @@ const EnvironmentLogs = ({ name, annotations }: EnvironmentLogsProps) => {
  * @param {function} toggleLogs - toggle logs visibility and fetch logs on show
  * @param {object} logs - log object from redux store enhanced with `show` property
  * @param {string} name - server name
- * @param {object} annotations - list of annotations
+ * @param {ReactNode | string} title - modal title
  */
 interface EnvironmentLogsPresentProps {
-  annotations: Record<string, unknown>;
+  title: ReactNode | string;
```
Comment (leafty) [inline client/src/components/Logs.tsx:373]:
```suggestion
  title: ReactNode;
```

`string` is already part of `ReactNode`.

Comment (andre-code) [reply client/src/components/Logs.tsx]: @leafty  fixed in the latest commit

**Comment by andre-code** [comment]
Note:
🐞 Custom environments are currently being created as GLOBAL. This issue is awaiting a fix in the backend.

**Comment by leafty** [comment]
The UI seems OK, but there are still issues with starting sessions, e.g. 
Screenshot_20241106-124105

**Review by leafty — APPROVED** [review]
LGTM

Please, do not merge until the whole feature is ready.

**Review by leafty — APPROVED** [review]

**Pull request merged**